### PR TITLE
Initial commit with previous sketch code...

### PR DESCRIPTION
**Pull request by TurkeyMan**
This is just a dump from our sketch-pad. I'm not attached to anything in here,

### CODE REVIEW

**Comment by p0nce** [comment]
About aspect ratio specifically : in image formats it is usually stored within vertical and horizontal DPI.
This gives a "Display Aspect Ratio" which is the aspect ratio of the image if it were display. 

The following formulas gives a "Pixel Aspect Ratio", there is also a "Sample Aspect Ratio" which can be forgotten.

```
PAR (pixel aspect ratio) = aspect ratio of a single pixel = width of a pixel / height of a pixel
SAR (sample aspect ratio) = width in stored pixels / height in stored pixels
DAR = width of the displayed image / height of the displayed image

// ALL "ASPECT RATIOS" ARE RELATED IN THIS WAY
DAR  = SAR * PAR
```

My point is that it's enough to store "vertical DPI" + "horizontal DPI" which give DAR + DPI scale factor combined in one.

**Comment by TurkeyMan** [comment]
The thing I don't like about using DPI for aspect ratio, is that most images don't have a reasonable DPI.
For a sans-DPI images, should the values be set to `DPI = [pixelAspect, 1]`? Then `pixelAspect == DPIx/DPIy`.

I don't really like this. much prefer just having the one value, since it's the overwhelmingly more common case then images with physical units.
Perhaps it could be:
```D
float pixelAspect;
float dpi;

float[2] getDPI() { return [ dpi, dpi/pixelAspect]; }
```
With that design, if no physical units exist, then dpi can just be `0`, but you don't lost the aspect ratio.
If your source data has `DPIx:DPIx`, then you can easily populate this struct.

File: src/wg/image/metadata.d [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// common metadata block for common but still somewhat niche data
+
+struct CommonMetadata
+{
+    enum char[4] ID = "META";
+
+    float pixelAspect = 1.0f; // defined as w/h; animorphic images > 1.0
+    float horizDpi = 0; // image has no associated physical units
+
+    float[2] getDPI() const { return [horizDpi, horizDpi / pixelAspect]; }
```
Comment (TurkeyMan) [inline src/wg/image/metadata.d]: How are we feeling about this?
I think it better expresses images that have no physical units associated; there's implicit knowledge available.

Comment (TurkeyMan) [reply src/wg/image/metadata.d]: The alternative way to express it with DPIx:DPIy, the DPI needs to have junk numbers to calculate the pixel aspect, even when no DPI exists.

Comment (p0nce) [reply src/wg/image/metadata.d]: I really like your solution:
```
    float pixelAspect = 1.0f; // defined as w/h; animorphic images > 1.0
    float horizDpi = 0; // image has no associated physical units
```

**Comment by p0nce** [comment]
If no DPI information exist, I think it's safe to assume 300 or 600, something like that.

**Comment by TurkeyMan** [comment]
I think it's ideal to be able to specify "no DPI" and let the user make their own decisions... that's the advantage of this solution, it works, even when DPI == 0.

**Pull request merged**